presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java

```diff
@@ -106,6 +113,34 @@ public ExecutingStatementResource(
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+
+        queryPurger.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        for (Entry<QueryId, Query> entry : queries.entrySet()) {
+                            // forget about this query if the query manager is no longer tracking it
+                            try {
+                                queryManager.getQueryInfo(entry.getKey());
+                            }
+                            catch (NoSuchElementException e) {
+                                // query is no longer registered
+                                queries.remove(entry.getKey());
+                            }
+                        }
+                    }
+                    catch (Throwable e) {
+                        log.warn(e, "Error removing old queries");
+                    }
+                },
+                200,
+                200,
```

Comment:
Does it have to be frequent?

Reply:
This is a copy of the purger from `QueuedStatementResource`... this basically restores the old behavior